libs/npm/core/src/components/table/FlexboxTable.tsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+
+import { Text } from '../form';
+import { ITableProps, SortFlag, TablePager, useTable } from '.';
+import * as styled from './styled';
+
+export const FlexboxTable = <T extends object>({
+  rowId,
+  columns,
+  data,
+  ...rest
+}: ITableProps<T>) => {
+  const table = useTable({
+    rowId,
+    columns,
+    data,
+    options: {
+      isMulti: rest.isMulti,
+      onRowClick: rest.onRowClick,
+      onCellClick: rest.onCellClick,
+      onColumnClick: rest.onColumnClick,
+      onSelectedChanged: rest.onSelectedChanged,
+      stopPropagation: rest.stopPropagation,
+      showHeader: rest.showHeader,
+      showFooter: rest.showFooter,
+      selectedRowIds: rest.selectedRowIds,
+      activeRowId: rest.activeRowId,
+    },
+    paging: {
+      manualPaging: rest.manualPaging,
+      pageIndex: rest.pageIndex,
+      pageSize: rest.pageSize,
+      pageCount: rest.pageCount,
+      pageButtons: rest.pageButtons,
+      showPaging: rest.showPaging,
+      scrollSize: rest.scrollSize,
+      onPageChange: rest.onPageChange,
+    },
+    sorting: {
+      showSort: rest.showSort,
+      sortOrder: rest.sortOrder,
+      onSortChange: rest.onSortChange,
+    },
+    filter: {
+      showFilter: rest.showFilter,
+      search: rest.search,
+      onFilterChange: rest.onFilterChange,
+    },
+    grouping: {
+      groupBy: rest.groupBy,
+      groupHeading: rest.groupHeading,
+    },
+  });
+  const [search, setSearch] = React.useState(table.search);
+
+  const style = {
+    className: `table${rest.className ? ` ${rest.className}` : ''}`,
+    columns: table.columns,
+    scrollSize: table.scrollSize,
+  };
+
+  return (
+    <styled.FlexboxTable {...style}>
+      {table.showFilter && (
+        <div className="filter">
+          <div>
+            <Text
+              name="filter"
+              placeholder="Search by keyword"
+              value={search}
+              onChange={(e) => {
+                setSearch(e.target.value);
+                table.applyFilter(e.target.value);
+              }}
+            />
+          </div>
+        </div>
+      )}
+      {table.options.showHeader && (
```

Comment:
I'd like to extend the functionality of the hook to include ways to fetch all the attributes similar to react-table.  It would make it easier to create new implementations.